src/main/fc/fc_rc.c

```diff
@@ -169,7 +169,9 @@ static void calculateSetpointRate(int axis)
 
         angleRate = applyRates(axis, rcCommandf, rcCommandfAbs);
     }
-    setpointRate[axis] = constrainf(angleRate, -SETPOINT_RATE_LIMIT, SETPOINT_RATE_LIMIT); // Rate limit protection (deg/sec)
+    // Rate limit from profile (deg/sec)
+    BUILD_BUG_ON(CONTROL_RATE_CONFIG_RATE_LIMIT_MAX > SETPOINT_RATE_LIMIT); // ensure CLI limits are <= hard-coded protection
```

Comment:
This will hopefully be optimised out (if the condition is false), but I think in a logical sense this should go into `initRcProcessing()` where the one-off initialisation is done.

Reply:
@mikeller : `BUILD_BUG_ON` is implemented as `(void)sizeof(...);`, it should not emit any code.

there is also `STATIC_ASSERT` which can be used outside of function, so test can be moved close to CONTROL_RATE_CONFIG_RATE_LIMIT_MAX definition

Reply:
@ledvinap: Good point about the `STATIC_ASSERT`, it seems to be a better match for this case.

Reply:
Either reimplement `STATIC_ASSERT` using `enum` instead of `typedef` (untested)
```
#define STATIC_ASSERT(condition, name) \ 
  enum { assert_failed_ ## name = sizeof(char [(condition) ? 1 : -1 ]) };
```

Or gcc contains C11 `_Static_assert` , so rewrite STATIC_ASSERT using it
```
#define STATIC_ASSERT(condition, name) _Static_assert((condition), #name)

Reply:
@ledvinap: I can add that to #6430.

Reply:
@ianrmurphy: Sorry it took so long, but STATIC_ASSERT is now all new and C11-shiny.